### PR TITLE
Solve #62

### PR DESCRIPTION
**Pull request by matthieugomez**
Allow not_in_place / n_ary + autodiff.

### CODE REVIEW

**Comment by spencerlyon2** [comment]
Looks good to me. What do you think @KristofferC ?

File: src/utils.jl [hunk 1]
```diff
@@ -39,4 +39,40 @@ function check_isfinite(x::Vector)
...
+function n_ary(f::Function)
+    f!(x::Vector, fx::AbstractArray) = copy!(fx, [f(x...)... ])
+end
+
+function not_in_place(f::Function)
+    f!(x::Vector, y::AbstractArray) = copy!(y, f(x))
+end
+
+function not_in_place(f::Function, g::Function)
+    depwarn("nlsolve(not_in_place(f, g)) is deprecated. Use nlsolve(not_in_place(f), not_in_place(g))")
```
Comment (KristofferC) [inline src/utils.jl]: Couldn't this function call what the deprecation is suggesting?

**Comment by KristofferC** [comment]
Seems ok, no idea how many people use the not in place `fg` though. Probably not many.

File: src/utils.jl (hunk 1 above)
Comment (matthieugomez) [inline src/utils.jl]: not exactly because one the replacement only works within nlsolve but I did change the function a little bit.

**Pull request closed without merging**